tests/test_transforms/test_embeddings/test_embedding_window_transform.py

```diff
@@ -0,0 +1,174 @@
+    lag_transforms = [
+        LagTransform(in_column=f"embedding_window_{i}", lags=[7], out_column=f"lag_{i}") for i in range(output_dims)
+    ]
+    filter_transforms = FilterFeaturesTransform(exclude=[f"embedding_window_{i}" for i in range(output_dims)])
+    transforms = [emb_transform] + lag_transforms + [filter_transforms]
+
+    pipeline = Pipeline(model=LinearMultiSegmentModel(), transforms=transforms, horizon=7)
+    pipeline.fit(example_tsds).forecast()
+
+
+@pytest.mark.parametrize(
+    "embedding_model",
+    [
+        TS2VecEmbeddingModel(input_dims=1, n_epochs=1, output_dims=2),
+    ],
+)
+@pytest.mark.smoke
+def test_backtest(example_tsds, embedding_model):
+    emb_transform = EmbeddingWindowTransform(
+        in_columns=["target"], embedding_model=embedding_model, out_column="embedding_window"
+    )
+    output_dims = embedding_model.output_dims
+    lag_transforms = [
+        LagTransform(in_column=f"embedding_window_{i}", lags=[7], out_column=f"lag_{i}") for i in range(output_dims)
+    ]
+    filter_transforms = FilterFeaturesTransform(exclude=[f"embedding_window_{i}" for i in range(output_dims)])
+    transforms = [emb_transform] + lag_transforms + [filter_transforms]
+
+    pipeline = Pipeline(model=LinearMultiSegmentModel(), transforms=transforms, horizon=7)
+    pipeline.backtest(ts=example_tsds, metrics=[SMAPE()], n_folds=2, n_jobs=2, joblib_params=dict(backend="loky"))
+
+
+@pytest.mark.parametrize(
+    "embedding_model",
+    [
+        TS2VecEmbeddingModel(input_dims=3, n_epochs=1),
+    ],
+)
+@pytest.mark.smoke
+def test_save(ts_with_exog_nan_begin, tmp_path, embedding_model):
+    transform = EmbeddingWindowTransform(in_columns=["target", "exog_1", "exog_2"], embedding_model=embedding_model)
+    transform.fit(ts=ts_with_exog_nan_begin)
+
+    path = pathlib.Path(tmp_path) / "tmp.zip"
+    transform.save(path=path)
+
+
+@pytest.mark.parametrize(
+    "embedding_model",
+    [
+        TS2VecEmbeddingModel(input_dims=3, n_epochs=1),
+    ],
+)
+@pytest.mark.smoke
+def test_load(ts_with_exog_nan_begin, tmp_path, embedding_model):
+    transform = EmbeddingWindowTransform(in_columns=["target", "exog_1", "exog_2"], embedding_model=embedding_model)
+    transform.fit(ts=ts_with_exog_nan_begin)
+
+    path = pathlib.Path(tmp_path) / "tmp.zip"
+    transform.save(path=path)
+    EmbeddingWindowTransform.load(path=path)
+
+
+@pytest.mark.parametrize(
+    "output_dims, out_column, expected_out_columns",
+    [(2, "emb", ["emb_0", "emb_1"]), (3, "lag", ["lag_0", "lag_1", "lag_2"])],
+)
+def test_get_out_columns(output_dims, out_column, expected_out_columns):
+    transform = EmbeddingWindowTransform(
+        in_columns=Mock(), embedding_model=Mock(output_dims=output_dims), out_column=out_column
+    )
+    assert sorted(expected_out_columns) == sorted(transform._get_out_columns())
+
+
+@pytest.mark.parametrize("embedding_model", [TS2VecEmbeddingModel(input_dims=3, n_epochs=1)])
```

Comment:
I think this test doesn't prove anything about not updating state. It looks just like that sequential fits produce the same result.